File: src/main/kotlin/coraythan/keyswap/users/UserProfile.kt
package coraythan.keyswap.users

import coraythan.keyswap.generatets.GenerateTs
import coraythan.keyswap.generic.Country
import coraythan.keyswap.users.search.UserSearchResult
import java.util.*

@GenerateTs
data class UserProfile(
        val id: UUID,
        val username: String,
        val email: String?,
        val sellerEmail: String?,
        val discord: String?,
        val tcoUsername: String?,
        val publicContactInfo: String?,
        val allowUsersToSeeDeckOwnership: Boolean,
        val country: Country?,
        val preferredCountries: List<Country>? = null,
        val lastVersionSeen: String?,
        val searchResult: UserSearchResult?,
        val allowsMessages: Boolean,
        val viewFutureSas: Boolean,
)

@GenerateTs
data class UserProfileUpdate(
        val email: String?,
        val publicContactInfo: String?,
        val allowsTrades: Boolean,
        val allowUsersToSeeDeckOwnership: Boolean,
        val currencySymbol: String,
        val country: Country?,
        val preferredCountries: List<Country>? = null,
        val sellerEmail: String? = null,
        val discord: String? = null,
        val tcoUsername: String? = null,
        val storeName: String? = null,
        val shippingCost: String? = null,
        val autoRenewListings: Boolean,
        val allowsMessages: Boolean,
        val viewFutureSas: Boolean,
)

File: ui/src/user/UserProfile.ts
import { UserSearchResult } from "./search/UserSearchResult"

export interface UserProfile {
    id: string
    username: string
    email?: string
    publicContactInfo?: string
    allowUsersToSeeDeckOwnership: boolean
    country?: string
    preferredCountries?: string[]
    searchResult?: UserSearchResult
}

export interface UserProfileUpdate {
    email?: string
    publicContactInfo?: string
    allowsTrades: boolean
    allowUsersToSeeDeckOwnership: boolean
    currencySymbol: string
    country?: string
    preferredCountries?: string[]
    sellerEmail?: string
    discord?: string
    storeName?: string
    shippingCost?: string
    autoRenewListings: boolean
}
